modules/protect/shared-functions.php

```diff
@@ -89,7 +142,14 @@ function jetpack_protect_save_whitelist( $whitelist ) {
 		return new WP_Error( 'invalid_ip', __( 'One of your IP addresses was not valid.', 'jetpack' ) );
 	}
 
-	update_site_option( 'jetpack_protect_whitelist', $new_items );
+	if ( $global ) {
+		update_site_option( 'jetpack_protect_network_whitelist', $new_items );
+		// once a user has saved their network wide whitelist, we can permanently remove the legacy option
+		delete_site_option( 'jetpack_protect_whitelist' );
```

Comment:
Since you're only saving the new items, this clears all legacy whitelists on sub-sites if even just one IP is saved globally.  Not ideal if I already had an extensive whitelist on my subsites.